stdlib/src/main/scala/chisel3/std/SBox.scala

```diff
@@ -0,0 +1,344 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+//  This is translated from https://github.com/riscv/riscv-crypto
+//
+//    2020-01-29    Markku-Juhani O. Saarinen <mjos@pqshield.com>
+//    Copyright (c) 2020, PQShield Ltd. All rights reserved.
+//    Converted to Chisel in 2022
```

Comment:
We should get some clarification on what the actual license of the RTL implementation is. Some files in the RTL directory are "do whatever you want" whereas this one doesn't specify. That may then implicitly default to the top-level CC 4.0 license which is incompatible with Apache.

I think this is just oversight as the `riscv/riscv-crypto` repo is supposed to be using CC 4.0 for all RISC-V related spec stuff, but I don't think the intent is to apply this to any RTL or software as well.

Reply:
Clarifying that the "do whatever you want" of other RTL files I spot-checked are ISC. This should be permissive enough to sweep it in.

Yes, creating an issue on `riscv/riscv-crypto` would be a good idea.

Reply:
> Yes, creating an issue on `riscv/riscv-crypto` would be a good idea.

Created. https://github.com/riscv/riscv-crypto/issues/159

Reply:
For what it's worth I'm making a public domain dedication for the sbox code. You have my permission to change the header accordingly. Of course, an acknowledgment of original authorship would be nice.

Background: RISC-V contributions are BSD 2-Clause (it's in the RISC-V membership agreement) and that was the license in the research repo where the S-Boxes came from https://github.com/mjosaarinen/lwaes_isa/blob/master/hdl/sboxes.v  So there always was a free license to use the copyrighted code.